File: dbfit-2.x/dbfit-core/src/main/java/dbfit/util/NameNormaliser.java
package dbfit.util;

/**
 * Utility class for normalising identifiers.
 */
public class NameNormaliser {
	private NameNormaliser() {
		// utility classes should not be instanciated
	}

	private static String replaceIllegalCharactersWithSpacesRegex = "[^a-zA-Z0-9_.#]";

	private static String replaceIllegalCharacters(final String name) {
		return name.replaceAll(replaceIllegalCharactersWithSpacesRegex, "");
	}

	public static String normaliseName(final String name) {
		if (name == null)
			return "";
		return replaceIllegalCharacters(name.toLowerCase());
	}
}

File: dbfit-dotnet/core/src/util/NameNormaliser.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.RegularExpressions;

namespace dbfit.util {
	public class NameNormaliser {
		private static Regex replaceIllegalCharactersWithSpacesRegex = new Regex(@"[^a-zA-Z0-9_.#]");
		private static string ReplaceIllegalCharacters(string name) {
			return replaceIllegalCharactersWithSpacesRegex.Replace(name, "");
		}
		public static String NormaliseName(String name) {
			if (name == null) return "";
			return ReplaceIllegalCharacters(name.ToLower());
		}
	}
}
